### PR TITLE
Feature/734 compare controls across run checkpoints

### PR DESCRIPTION
**Pull request by DzMakatun**
New warning in monitoring: Control measures mismatch
CheckpointUtils: iterate over an array of checkpoints, compare control measures

### CODE REVIEW

File: menas/ui/service/MonitoringService.js [hunk 1]
```diff
@@ -296,8 +300,12 @@ var MonitoringService = new function() {
   };
 
   // find raw recordcount
-  // TODO: Compare controlls across all checkpoints
-  this.processCheckpoints = function(oRun) {
+  this.setRawRecordcount = function (oRun) {
+    let rawRecordcount = oRun["controlMeasure"]["metadata"]["additionalInfo"]["raw_record_count"];
+    if (rawRecordcount != null && !isNaN(rawRecordcount) ) {
+      oRun["rawRecordcount"] = rawRecordcount;
+      return;
```
Comment (Zejnilovic) [inline menas/ui/service/MonitoringService.js]: What is the point of this `return`?

File: menas/ui/service/MonitoringService.js [hunk 2]
```diff
@@ -352,9 +369,10 @@ var MonitoringService = new function() {
           oRun["warnings"] = [];
 
           MonitoringService.processLatestCheckpoint(oRun);
+          MonitoringService.processCeckpoints(oRun);
           MonitoringService.processRunStatus(oRun);
           MonitoringService.processDirSizes(oRun);
-          MonitoringService.processCheckpoints(oRun);
+          MonitoringService.setRawRecordcount(oRun);
```
Comment (Zejnilovic) [inline menas/ui/service/MonitoringService.js]: Atum speaks of `RecordCount` or `recordCount`
```suggestion
          MonitoringService.setRawRecordCount(oRun);
```

File: menas/ui/service/CheckpointUtils.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+class CheckpointUtils {
+
+  static checkpointsContlolsMatch(aCheckpoints) {
```
Comment (Zejnilovic) [inline menas/ui/service/CheckpointUtils.js]:
```suggestion
  static checkpointsControlsMatch(aCheckpoints) {
```

**Comment by DzMakatun** [comment]
> Even though all my checkpoints match I get a checkpoint mismatch
> image
> image

thnx, which test dataset is it?

File: menas/ui/service/MonitoringService.js (hunk 1 above)
Comment (DzMakatun) [reply menas/ui/service/MonitoringService.js]: If the raw_record_count is provided in run metadata (as in the new Enceladus version) we do not need to search it in the checkpoints, therefore, we exit the method here.

File: menas/ui/service/MonitoringService.js (hunk 2 above)
Comment (DzMakatun) [reply menas/ui/service/MonitoringService.js]: ok, thnx

File: menas/ui/service/CheckpointUtils.js (hunk 3 above)
Comment (DzMakatun) [reply menas/ui/service/CheckpointUtils.js]: 😆  thank you

**Comment by Zejnilovic** [comment]
> thnx, which test dataset is it?

 My custom one from the previous test of monitoring

**Comment by DzMakatun** [comment]
> Even though all my checkpoints match I get a checkpoint mismatch
> image
> image

Very nice catch. 

Checked the dataset simpleCSV. This is because the checkpoint controls are different in "controlCol" ("id" VS "*"). I will ease the controls comparison (compare "controlName" and "controlValue" only). _Anyone would like to comment on this idea?_

 ```
"checkpoints" : [ 
            {
                "name" : "Source",
                "processStartTime" : "01-06-2019 08:00:00",
                "processEndTime" : "01-06-2019 08:00:00",
                "workflowName" : "Source",
                "order" : 1,
                "controls" : [ 
                    {
                        "controlName" : "recordCount",
                        "controlType" : "controlType.Count",
                        "controlCol" : "id",
                        "controlValue" : "6"
                    }
                ]
            }, 
            {
                "name" : "Raw",
                "processStartTime" : "01-06-2019 08:00:00",
                "processEndTime" : "01-06-2019 08:00:00",
                "workflowName" : "Raw",
                "order" : 2,
                "controls" : [ 
                    {
                        "controlName" : "recordCount",
                        "controlType" : "controlType.Count",
                        "controlCol" : "*",
                        "controlValue" : "6"
                    }
                ]
            }
        ]
```

**Comment by Zejnilovic** [comment]
> Checked the dataset simpleCSV. This is because the checkpoint controls are different in "controlCol" ("id" VS "*"). I will ease the controls comparison (compare "controlName" and "controlValue" only). _Anyone would like to comment on this idea?_ @yruslan @Zejnilovic @lokm01

Agreed. Pair of name and value should be unique.

**Comment by DzMakatun** [comment]
#734

**Pull request merged**